### PR TITLE
Allow scalar surface orientation inputs to pvlib.bifacial.pvfactors_timeseries

### LINKED ISSUE #1127: bifacial.pvfactors expects arrays for surface_tilt and surface_azimuth inputs
The docstring for `bifacial.pvfactors` describes these two parameters:

```
    surface_azimuth: numeric
        Azimuth angle of the front surface of the PV modules, using pvlib's
        convention (deg)
    surface_tilt: numeric
        Tilt angle of the PV modules, going from 0 to 180 (deg)
```
Based on the docstring I would expect the function to run with e.g. `surface_azimuth` being a float. But it doesn't; it only runs when `surface_azimuth` is an array or Series of the same length as other time-dependent inputs. Providing `surface_azimuth` as a float yields `IndexError: too many indices for array`

Issue originally reported in #1125 

To reproduce:
```
from datetime import datetime
import numpy as np
import pandas as pd
import pvlib


df_inputs = pd.DataFrame({'solar_zenith': [20., 50.],
                          'solar_azimuth': [110., 250.],
                          'surface_tilt': [10., 20.],
                          'surface_azimuth': [90., 270.],
                          'dni': [1000., 900.],
                          'dhi': [50., 100.],
                          'albedo': [0.2, 0.2]},
                          index=[datetime(2017, 8, 31, 11),
                                 datetime(2017, 8, 31, 15)])

pvarray_parameters = {'n_pvrows': 3,            # number of pv rows
                      'pvrow_height': 1,        # height of pvrows (measured at center / torque tube)
                      'pvrow_width': 1,         # width of pvrows
                      'axis_azimuth': 0.,       # azimuth angle of rotation axis
                      'gcr': 0.4,               # ground coverage ratio
                      }

pvfactor = pvlib.bifacial.pvfactors_timeseries(
    solar_azimuth=df_inputs['solar_azimuth'],
    solar_zenith=df_inputs['solar_zenith'],
    surface_azimuth=180, 
    surface_tilt=40, 
    axis_azimuth=0.0,
    timestamps=df_inputs.index, 
    dni=df_inputs['dni'],
    dhi=df_inputs['dhi'],
    gcr=2.0/7.0,
    pvrow_height=1,
    pvrow_width=7,
    albedo=0.2,
    n_pvrows=2,
    index_observed_pvrow=3,
    rho_front_pvrow=0.03,
    rho_back_pvrow=0.05,
    horizon_band_angle=15)
```

### LINKED ISSUE #1332: pvfactors_timeseries docstring types are incorrect
**Describe the bug**
The [pvfactors_timeseries](https://pvlib-python.readthedocs.io/en/v0.9.0/generated/pvlib.bifacial.pvfactors_timeseries.html) docstring currently lists several parameters as "numeric", implying scalar values are acceptable ([docs](https://pvlib-python.readthedocs.io/en/stable/contributing.html#documentation)).  However, some parameters cannot actually be scalars and should be listed as "array-like" instead.

**To Reproduce**
Example showing that scalar inputs for the module orientation parameters are not acceptable:

<details>
  <summary>Click to expand!</summary>

```python
import pvlib
import pandas as pd
from pvlib.bifacial import pvfactors_timeseries

times = pd.date_range('2019-01-01', '2019-01-02', freq='15min').tz_localize('Etc/GMT+5')
location = pvlib.location.Location(40, -80)
solarpos = location.get_solarposition(times)
clearsky = location.get_clearsky(times)

kwargs = dict(
    solar_azimuth=solarpos['azimuth'],
    solar_zenith=solarpos['apparent_zenith'],
    axis_azimuth=0,
    timestamps=times,
    dni=clearsky['dni'],
    dhi=clearsky['dhi'],
    gcr=0.3,
    pvrow_height=1.5,
    pvrow_width=1.0,
    albedo=0.2
)

pvfactors_timeseries(surface_tilt=20, surface_azimuth=90, **kwargs)  # IndexError: too many indices for array
pvfactors_timeseries(surface_tilt=pd.Series(20, index=times),
                     surface_azimuth=pd.Series(90, index=times),
                     **kwargs)  # works
```
</details>

**Expected behavior**
Correct the types in the docstring to accurately reflect pvfactors' requirements. 

**Additional context**
Originally discussed here: https://github.com/pvlib/pvlib-python/discussions/1331#discussioncomment-1542801
Edit: this is more or less a duplicate of #1127

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] Closes #1332, closes #1127
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - ~Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

As mentioned in #1332, I opted to rework the function to allow scalar orientation inputs instead of editing the docstring to prohibit them.  I also did some cleanup in the tests to avoid code duplication.

### CODE REVIEW

**Review by cwhanse — APPROVED** [review]

File: pvlib/bifacial.py [hunk 1]
```diff
@@ -82,31 +82,18 @@ def pvfactors_timeseries(
...
-        dhi = dhi.values
-    elif isinstance(dhi, list):
-        dhi = np.array(dhi)
+    # Convert Series, list, float inputs to numpy arrays
+    solar_azimuth = np.array(solar_azimuth)
+    solar_zenith = np.array(solar_zenith)
+    dni = np.array(dni)
+    dhi = np.array(dhi)
+    # GH 1127, GH 1332
+    if np.isscalar(surface_tilt):
```
Comment (cwhanse) [inline pvlib/bifacial.py:91]: Use `numpy.atleast_1d`? It would avoid the `if` but otherwise I don't know if it's better.

Comment (kanderso-nrel) [reply pvlib/bifacial.py:91]: I think the requirement is that `surface_tilt` be an array of the same length as the meteo inputs, so `np.atleast_1d` would still need to be followed by a conditional `np.repeat` or something.

Comment (wholmgren) [reply pvlib/bifacial.py:91]: `full_like` docs say it requires a scalar input, but this works:

```
a = np.array([1,2,3])

np.full_like(a, 7)

array([7, 7, 7])

np.full_like(a, np.array([7,8,9]))

array([7, 8, 9])
```

**Review by wholmgren — APPROVED** [review]

File: pvlib/bifacial.py (hunk 1 above)
Comment (kanderso-nrel) [reply pvlib/bifacial.py]: Seems to work with lists and Series too.  Always happy to learn a new numpy function :)

**Comment by kanderso-nrel** [comment]
New test failures are the same as https://github.com/pvlib/pvlib-python/pull/717#issuecomment-1004215937, so ignoring here

**Pull request merged**